src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java

```diff
@@ -24,25 +24,49 @@
 import org.bukkit.entity.Player;
 import org.eclipse.jdt.annotation.Nullable;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.UUID;
+
 @Name("Player Protocol Version")
 @Description("Player's protocol version. For more information and list of protocol versions <a href='https://wiki.vg/Protocol_version_numbers'>visit wiki.vg</a>.")
 @Examples({"command /protocolversion &ltplayer&gt:",
 	"\ttrigger:",
 	"\t\tsend \"Protocol version of %arg-1%: %protocol version of arg-1%\""})
-@Since("2.6.2")
+@Since("2.6.2, INSERT VERSION ViaVersion support")
 @RequiredPlugins("Paper 1.12.2 or newer")
 public class ExprPlayerProtocolVersion extends SimplePropertyExpression<Player, Integer> {
 
+	private static final boolean VIAVERSION_EXISTS = Skript.classExists("com.viaversion.viaversion.api.ViaAPI");
+	@Nullable
+	private static Object VIA_API;
+
 	static {
-		if (Skript.classExists("com.destroystokyo.paper.network.NetworkClient")) {
+		if (Skript.classExists("com.destroystokyo.paper.network.NetworkClient") || VIAVERSION_EXISTS) {
 			register(ExprPlayerProtocolVersion.class, Integer.class, "protocol version", "players");
 		}
+
+		if (VIAVERSION_EXISTS) {
+			try {
+				VIA_API = Class.forName("com.viaversion.viaversion.api.Via")
+					.getDeclaredMethod("getAPI")
+					.invoke(null);
+			} catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException | IllegalAccessException ignored) {}
+		}
 	}
 
 	@Override
 	@Nullable
 	public Integer convert(Player player) {
-		int version = player.getProtocolVersion();
+		int version = -1;
+		if (VIAVERSION_EXISTS && VIA_API != null) {
+			try {
+				Method getPlayerVersion = VIA_API.getClass().getDeclaredMethod("getPlayerVersion", UUID.class);
```

Comment:
Store this method during clinit instead of obtaining a new handle each time. Method handles get inlined after their first use, so we're just wasting efficiency by getting it again each time.